### PR TITLE
Document the default build environment variables

### PR DESCRIPTION
**Pull request by kit-ty-kate**
~Should be merged after https://github.com/ocaml/opam/pull/4492~

### CODE REVIEW

File: doc/pages/Manual.md [hunk 1]
```diff
@@ -1077,6 +1077,15 @@ files.
 - <a id="opamfield-build-env">`build-env: [ <environment-update> ... ]`</a>:
   defines environment updates that will be applied when running the package's
   build, install and remove scripts.
+  
+  By default, the following environment variables are set by opam:
```
Comment (dra27) [inline doc/pages/Manual.md:1081]:
```suggestion
  The following environment variables are set by opam (but can be overridden by `build-env`):
```

File: doc/pages/Manual.md [hunk 2]
```diff
@@ -1077,6 +1077,15 @@ files.
...
+  + `CDPATH=`
+  + `MAKEFLAGS=`
+  + `MAKELEVEL=`
+  + `OPAM_PACKAGE_NAME=<pkg>` (with `<pkg>` being the name of the current package being built/installed/removed)
+  + `OPAM_PACKAGE_VERSION=<ver>` (with `<ver>` being the version of the current package being built/installed/removed)
```
Comment (dra27) [inline doc/pages/Manual.md:1086]:
```suggestion
  + `OPAM_PACKAGE_NAME=<pkg>` (`<pkg>` is the name of the package being built/installed/removed)
  + `OPAM_PACKAGE_VERSION=<ver>` (`<ver>` is the version of the package being built/installed/removed)
```
(slightly more terse)

File: doc/pages/Manual.md [hunk 3]
```diff
@@ -1077,6 +1077,15 @@ files.
...
+  + `OPAMCLI=2.0` (since opam 2.1)
+  + `TMP` and `TMPDIR` are also set by the sandbox (bubblewrap) script but are not to be relied upon.
```
Comment (dra27) [inline doc/pages/Manual.md:1088]: I'd reword this slightly to explain why:
```suggestion
  + `TMP` and `TMPDIR` are set by the sandbox script (bubblewrap), but should not be relied on since ...
```

**Review by dra27 — COMMENTED** [review]
A couple of tweak suggestions and I think the `TMP`/`TMPDIR` bit should expanded (in general I think it's better always to say _why_ you shouldn't do something)

**Comment by AltGr** [comment]
Please make sure the markdown is correct with `make doc`, then checking `doc/pages/Manual.html` (this part hasn't been dune-ified yet). Thanks !

File: doc/pages/Manual.md (hunk 3 above)
Comment (kit-ty-kate) [reply doc/pages/Manual.md:1088]:
```suggestion
  + `TMP` and `TMPDIR` are set by the sandbox script (bubblewrap), but should not be relied on since the sandbox is not used on every platforms and can be disabled by the user.
```
What about this?

Comment (dra27) [reply doc/pages/Manual.md:1088]: LGTM!

**Comment by kit-ty-kate** [comment]
Here is what the modified section looks like when compiled:
scrn-2021-01-13-14-54-49

Looks good to go.

File: doc/pages/Manual.md [hunk 4]
```diff
@@ -1078,6 +1078,15 @@ files.
...
   build, install and remove scripts.
 
+    The following environment variables are set by opam (but can be overridden by `build-env`):
+    - `CDPATH=`
+    - `MAKEFLAGS=`
+    - `MAKELEVEL=`
+    - `OPAM_PACKAGE_NAME=<pkg>` (`<pkg>` is the name of the package being built/installed/removed)
+    - `OPAM_PACKAGE_VERSION=<ver>` (`<ver>` is the version of the package being built/installed/removed)
+    - `OPAMCLI=2.0` (since opam 2.1)
+    - `TMP` and `TMPDIR` are set by the sandbox script (bubblewrap), but should not be relied on since the sandbox is not used on every platforms and can be disabled by the user.
```
Comment (dra27) [inline doc/pages/Manual.md:1088]:
```suggestion
    - `TMP` and `TMPDIR` are set by the sandbox script (bubblewrap), but should not be relied on since the sandbox is not used on all platforms and can be disabled by the user.
```

**Review by dra27 — COMMENTED** [review]
Oops - tiny nit in the text but otherwise looks (both in Markdown and HTML) good to go, yes!

**Comment by kit-ty-kate** [comment]
(feel free to squash/rebase-merge instead the default-merge for this)

**Comment by rjbou** [comment]
Thanks!
